docs/basic-features/eslint.md

```diff
@@ -0,0 +1,113 @@
+---
+description: Next.js supports ESLint by default. You can get started with ESLint in Next.js here.
+---
+
+# ESLint
+
+Since version **11.0.0**, Next.js provides an integrated [ESLint](https://eslint.org/) experience out of the box. To get started, run `next lint`:
+
+```bash
+next lint
+```
+
+If you don't already have ESLint configured in your application, you will be guided through the installation of any required packages.
+
+```bash
+next lint
+
+# You'll see instructions like these:
+#
+# Please install eslint and eslint-config-next by running:
+#
+#         yarn add --dev eslint eslint-config-next
+#
+# ...
+```
+
+If no ESLint configuration is present, Next.js will create an `.eslintrc` file in the root of your project and automatically configure it with the base configuration:
+
+```
+{
+  "extends": "next"
+}
+```
+
+Now you can run `next lint` every time you want to run ESLint to catch errors
+
+> The default base configuration (`"extends": "next"`) can be updated at any time and will only be included if no ESLint configuration is present.
+
+We recommend using an appropriate [integration](https://eslint.org/docs/user-guide/integrations#editors) to view warnings and errors directly in your code editor during development.
+
+## Linting During Builds
+
+Once ESLint has been set up, it will automatically run during every build (`next build`). Errors will fail the build while warnings will not.
+
+If you do not want ESLint to run as a build step, it can be disabled using the `--no-lint` flag:
+
+```bash
+next build --no-lint
+```
+
+This is not recommended unless you have configured ESLint to run in a separate part of your workflow (for example, in CI or a pre-commit hook).
+
+## Linting Custom Directories
+
+By default, Next.js will only run ESLint for all files in the `pages/` directory. However, you can specify other custom directories to run by using the `--dir` flag in `next lint`:
+
+```bash
+next lint --dir components --dir lib
+```
+
+## ESLint Plugin
+
+Next.js provides an ESLint plugin, [`eslint-plugin-next`](https://www.npmjs.com/package/@next/eslint-plugin-next), that makes it easier to catch common issues and problems in a Next.js application. The full set of rules can be found in the [package repository](https://github.com/vercel/next.js/tree/master/packages/eslint-plugin-next/lib/rules).
```

Comment:
Instead of linking to them we should maintain a list in this document, but for now it's fine to have it linked 👍